pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -512,6 +514,9 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 	if err != nil {
 		return statusUnchanged, err
 	}
+	if job == nil {
+		return statusUnchanged, errors.Errorf("Could not find job %s", app.Status.JobStatus.JobID)
```

Comment:
Not directly related to the change in the PR, but I'm curious how we end up here. It looks like `GetJobForApplication` only returns a `nil` job when the jobID is empty. If jobID were empty we should end up in the `if app.Status.JobStatus.JobID == "" ` above and not here? Basically curious, how you ran into the nil deference issue and if there's another bug that's being masked. We can discuss outside of the PR too :)

Reply:
`flinkClient.GetJobOverview` has this error handling logic:

```go
var jobOverviewResponse FlinkJobOverview
if err = json.Unmarshal(response.Body(), &jobOverviewResponse); err != nil {
	logger.Errorf(ctx, "Failed to unmarshal FlinkJob %v, err %v", response, err)
}

return &jobOverviewResponse, nil
```

Which means that if we fail to unmarshal the response, we end up getting `nil, nil`.

Reply:
Actually, that explanation is wrong—we'll get an empty FlinkJobOverview struct, not nil. I think your'e right that the check should not be needed here.

Reply:
Ah okay, that makes sense. Thanks for the context. :) Yeah, maybe `GetJobOverview`'s error response should be changed..